### ISSUE TITLE
Move ASV benchmark related files to own folder

### ISSUE THREAD

**Issue by Illviljan** [scikit-image/scikit-image#5591]
It's rather common for asv benchmark related files to be located in it's own folder, see 
* https://github.com/numpy/numpy
* https://github.com/pandas-dev/pandas
* https://github.com/pydata/xarray

The suggestion is to align with the other repos and move the `asv.conf.json` and `benchmarks` folder to it's own `asv_bench` folder. This will likely require some changes in the 
* CI benchmark. See https://github.com/pydata/xarray/pull/5796 for the possible edits
* Some other parts of the documentation.

**Comment by grlee77** [comment]
PyWavelets also uses a `benchmarks` subfolder. It does make it a bit easier to locate the corresponding `asv.conf.json` and potentially a benchmark-specific README if they are located in a subfolder, but I don't have a very strong opinion about moving it vs. leaving it where it is. What do others think?

Certainly, there are many functions for which we are currently missing benchmarks altogether and we would welcome contributions there!

**Comment by jni** [comment]
My personal gut reaction so far is that the benefit (mild uniformity?) is not sufficient to justify the code churn. I don't like the `asv_bench` name as it is very tool-specific. New benchmarking tools might emerge e.g. maybe [conbench](https://ursalabs.org/blog/announcing-conbench/) gains momentum and we need to reorg again. Plus, as a newcomer to the repo, a `benchmarks` folder is much clearer and more self-descriptive than `asv_bench`.

**Comment by Illviljan** [comment]
My motivation for this change was the excellent asv workflow that I found through https://labs.quansight.org/blog/2021/08/github-actions-benchmarks/. 
Unfortunately the workflow required a few edits to get running in what I would consider the "normal" setup; inside a subfolder.

Feel free to call it `benchmarks` instead if you prefer, that's how numpy does it anyway and is an easier change inside the workflow. The difficult part for me was how to trigger the workflow correctly when the config file was inside a subfolder, and considering the blog post I suppose you want to make it easy for other projects to also use the workflow for benchmarks?

**Comment by grlee77** [comment]
> My motivation for this change was the excellent asv workflow that I found through https://labs.quansight.org/blog/2021/08/github-actions-benchmarks/ and #5424.

Indeed, we would like other projects to be able to easily adopt the approach. I am sure @jaimergp would be glad to hear that others are already using it! What project are you working on applying it to?

**Comment by Illviljan** [comment]
I've been adding it to xarray. See https://github.com/pydata/xarray/pull/5796 for the PR.
